presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfStripeCacheMode.java

```diff
@@ -19,14 +19,65 @@
 public enum DwrfStripeCacheMode
 {
     // no cache
-    NONE,
+    NONE(false, false),
 
     // cache contains stripe indexes
-    INDEX,
+    INDEX(true, false),
 
     // cache contains stripe footers
-    FOOTER,
+    FOOTER(false, true),
 
     // cache contains stripe indexes and footers
-    INDEX_AND_FOOTER;
+    INDEX_AND_FOOTER(true, true);
+
+    private final boolean hasIndex;
+    private final boolean hasFooter;
+
+    DwrfStripeCacheMode(boolean hasIndex, boolean hasFooter)
+    {
+        this.hasIndex = hasIndex;
+        this.hasFooter = hasFooter;
+    }
+
+    public boolean hasIndex()
+    {
+        return hasIndex;
+    }
+
+    public boolean hasFooter()
+    {
+        return hasFooter;
+    }
+
+    /**
+     * Get the position of an element in the cacheOffsets that can be used to
+     * get the offset for the index streams slice for a given stripe.
+     */
+    public int getIndexOffsetPosition(int stripeOrdinal)
+    {
+        switch (this) {
+            case INDEX:
+                return stripeOrdinal;
+            case INDEX_AND_FOOTER:
+                return stripeOrdinal * 2;
+            default:
+                throw new IllegalStateException("This mode doesn't support index cache: " + this);
```

Comment:
This class does not have toString implementation, but I assume enum has default toString implementation, just verifying that is the case here.

Reply:
Correct, by default enum.toString looks exactly the same as enum.name()